docs/aws/audit/elbmonitoring/rules/elb_check_unused_remediation.mdx

```diff
@@ -1,6 +1,286 @@
+   - This helps in identifying ELBs that are not being used and can be reviewed for potential decommissioning.
+
+4. **Periodic Audits:**
+   - Schedule regular audits (e.g., monthly or quarterly) to review all ELBs.
+   - During the audit, verify that each ELB is still required and actively serving traffic.
+   - Document the findings and take action on any ELBs that are identified as unused or unnecessary.
+
+By following these steps, you can proactively manage and prevent unused ELBs in your AWS environment.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent having unused Elastic Load Balancers (ELBs) in AWS using the AWS CLI, you can follow these steps:
+
+1. **List All ELBs:**
+   First, you need to list all the existing ELBs in your account to identify which ones are currently in use and which ones are not.
+
+   ```sh
+   aws elb describe-load-balancers --query 'LoadBalancerDescriptions[*].LoadBalancerName'
+   ```
+
+2. **Check ELB Usage:**
+   For each ELB, check if there are any instances registered with it. An ELB with no registered instances is likely unused.
+
+   ```sh
+   aws elb describe-load-balancers --load-balancer-names <load-balancer-name> --query 'LoadBalancerDescriptions[*].Instances'
+   ```
+
+3. **Tag ELBs:**
+   Tag your ELBs with metadata that indicates their purpose and owner. This helps in identifying and managing ELBs more effectively.
+
+   ```sh
+   aws elb add-tags --load-balancer-names <load-balancer-name> --tags Key=Purpose,Value=<purpose> Key=Owner,Value=<owner>
+   ```
+
+4. **Automate Monitoring:**
+   Set up a scheduled AWS Lambda function or a CloudWatch rule to periodically check for unused ELBs and notify the relevant stakeholders. This can be done using a Python script executed by AWS Lambda.
+
+   Example Python script snippet for Lambda:
+   ```python
+   import boto3
+
+   def lambda_handler(event, context):
+       client = boto3.client('elb')
+       elbs = client.describe_load_balancers()['LoadBalancerDescriptions']
+       unused_elbs = [elb['LoadBalancerName'] for elb in elbs if not elb['Instances']]
+       
+       if unused_elbs:
+           # Notify stakeholders or take action
+           print(f"Unused ELBs: {unused_elbs}")
+   ```
+
+By following these steps, you can ensure that you are actively monitoring and managing your ELBs to prevent having unused ones in your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent unused Elastic Load Balancers (ELBs) in AWS using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK (Boto3) and Authentication:**
+   - Install the Boto3 library if you haven't already.
+   - Configure your AWS credentials.
+
+   ```bash
+   pip install boto3
+   ```
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon EC2
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>